device/api/umd/device/wormhole_implementation.h

```diff
@@ -103,49 +103,44 @@ enum class arc_message_type {
 
 // DEVICE_DATA
 static const tt_xy_pair TENSIX_GRID_SIZE = {8, 10};
+// clang-format off
 static const std::vector<tt_xy_pair> TENSIX_CORES = {{
-    {1, 1},  {2, 1},  {3, 1},  {4, 1},  {6, 1},  {7, 1},  {8, 1},  {9, 1},  {1, 2},  {2, 2},  {3, 2},  {4, 2},
-    {6, 2},  {7, 2},  {8, 2},  {9, 2},  {1, 3},  {2, 3},  {3, 3},  {4, 3},  {6, 3},  {7, 3},  {8, 3},  {9, 3},
-    {1, 4},  {2, 4},  {3, 4},  {4, 4},  {6, 4},  {7, 4},  {8, 4},  {9, 4},  {1, 5},  {2, 5},  {3, 5},  {4, 5},
-    {6, 5},  {7, 5},  {8, 5},  {9, 5},  {1, 7},  {2, 7},  {3, 7},  {4, 7},  {6, 7},  {7, 7},  {8, 7},  {9, 7},
-    {1, 8},  {2, 8},  {3, 8},  {4, 8},  {6, 8},  {7, 8},  {8, 8},  {9, 8},  {1, 9},  {2, 9},  {3, 9},  {4, 9},
-    {6, 9},  {7, 9},  {8, 9},  {9, 9},  {1, 10}, {2, 10}, {3, 10}, {4, 10}, {6, 10}, {7, 10}, {8, 10}, {9, 10},
+    {1, 1},   {2, 1},  {3, 1},  {4, 1},  {6, 1},  {7, 1},  {8, 1},  {9, 1},
+    {1, 2},   {2, 2},  {3, 2},  {4, 2},  {6, 2},  {7, 2},  {8, 2},  {9, 2},
+    {1, 3},   {2, 3},  {3, 3},  {4, 3},  {6, 3},  {7, 3},  {8, 3},  {9, 3},
+    {1, 4},   {2, 4},  {3, 4},  {4, 4},  {6, 4},  {7, 4},  {8, 4},  {9, 4},
+    {1, 5},   {2, 5},  {3, 5},  {4, 5},  {6, 5},  {7, 5},  {8, 5},  {9, 5},
+    {1, 7},   {2, 7},  {3, 7},  {4, 7},  {6, 7},  {7, 7},  {8, 7},  {9, 7},
+    {1, 8},   {2, 8},  {3, 8},  {4, 8},  {6, 8},  {7, 8},  {8, 8},  {9, 8},
+    {1, 9},   {2, 9},  {3, 9},  {4, 9},  {6, 9},  {7, 9},  {8, 9},  {9, 9},
+    {1, 10}, {2, 10}, {3, 10}, {4, 10}, {6, 10}, {7, 10}, {8, 10}, {9, 10},
     {1, 11}, {2, 11}, {3, 11}, {4, 11}, {6, 11}, {7, 11}, {8, 11}, {9, 11},
 }};
+// clang-format on
 
 static const std::size_t NUM_DRAM_BANKS = 6;
 static const std::size_t NUM_NOC_PORTS_PER_DRAM_BANK = 3;
 static const tt_xy_pair DRAM_GRID_SIZE = {NUM_DRAM_BANKS, NUM_NOC_PORTS_PER_DRAM_BANK};
+// clang-format off
 static const std::vector<tt_xy_pair> DRAM_CORES = {
-    {{0, 0},
-     {5, 0},
-     {0, 1},
-     {5, 1},
-     {5, 2},
-     {5, 3},
-     {5, 4},
-     {0, 5},
-     {5, 5},
-     {0, 6},
-     {5, 6},
-     {0, 7},
-     {5, 7},
-     {5, 8},
-     {5, 9},
-     {5, 10},
-     {0, 11},
-     {5, 11}}};
+    {{0, 0}, {0, 1}, {0, 11},
+     {0, 5}, {0, 6},  {0, 7},
+     {5, 0}, {5, 1}, {5, 11},
+     {5, 2}, {5, 9}, {5, 10},
+     {5, 3}, {5, 4},  {5, 8},
+     {5, 5}, {5, 6},  {5, 7}}};
+// clang-format on
 
 // TODO: DRAM locations should be deleted. We keep it for compatibility with
 // the existing code in clients which rely on DRAM_LOCATIONS.
 static const std::vector<tt_xy_pair> DRAM_LOCATIONS = DRAM_CORES;
 
 static const tt_xy_pair ARC_GRID_SIZE = {1, 1};
-static const std::vector<tt_xy_pair> ARC_CORES = {{0, 2}};
+static const std::vector<tt_xy_pair> ARC_CORES = {{0, 10}};
```

Comment:
I'm afraid there might be more issues like this.
Why was this not caught by tests? I guess tests were written so that this passes :) Is there some other codepath which we know is working which we can test against? Or is just the set of constants used by the clients (tt-lens for example) just a different set of constants?

Reply:
Just a different set of constants. I went through the git blame for this one. The wrong constant in UMD were there for a long time, and soc descriptor coordinates (which were the one everyone was using) were different from the constants in implementation files. No one was using implementation constants until Coordinate manager, and it was properly working for the translation of the wrong constants, it was expecting these. The bug came when @tt-vjovanovic was using coordinate manager created with just arch, and was expecting to translate soc descriptor coordinates (which in theory should be the same). Since no one was using implementation coordinates, I think we are good. 

I checked other values as well